test/e2e/proxy_test.go

```diff
@@ -373,14 +402,35 @@ func nodePortTestCases(t *testing.T, data *TestData, portStrCluster, portStrLoca
 		testNodePortLocalFromRemote(t, data, nodes, reverseStrs(localUrls), nodeIPs, reverseStrs(hostnames))
 	})
 	t.Run("ExternalTrafficPolicy:Local/Client:Node", func(t *testing.T) {
-		testNodePortLocalFromNode(t, data, nodes, localUrls, hostnames)
+		expectedHostnames := make([]string, len(hostnames))
+		copy(expectedHostnames, hostnames)
+		if hostNetwork {
+			for idx := range expectedHostnames {
+				if nodeOS(nodes[idx]) == "windows" {
+					expectedHostnames[idx] = pauseContainerIDs[idx]
+				}
+			}
+		}
+		testNodePortLocalFromNode(t, data, nodes, localUrls, expectedHostnames)
 	})
 	t.Run("ExternalTrafficPolicy:Local/Client:Pod", func(t *testing.T) {
-		testNodePortLocalFromPod(t, data, pods, localUrls, podIPs, hostnames)
+		expectedClientIPs := make([]string, len(podIPs))
+		copy(expectedClientIPs, podIPs)
+		expectedHostnames := make([]string, len(hostnames))
+		copy(expectedHostnames, hostnames)
+		if hostNetwork {
+			for idx := range podOSes {
+				if podOSes[idx] == "windows" {
+					expectedClientIPs[idx] = nodeIPs[idx]
```

Comment:
isn't this already true? why is `expectedClientIPs` different for Windows?

Reply:
Added comments.